client/app/bundles/course/statistics/pages/StatisticsIndex/course/StudentPerformanceTable.tsx

```diff
@@ -0,0 +1,416 @@
+    defaultMessage: 'Experience Points',
+  },
+  learningRate: {
+    id: 'course.statistics.StatisticsIndex.course.StudentPerformanceTable.learningRate',
+    defaultMessage: 'Learning Rate',
+  },
+  numSubmissionsDetails: {
+    id: 'course.statistics.StatisticsIndex.course.StudentPerformanceTable.numSubmissionsDetails',
+    defaultMessage: 'No. of Submissions (Total: {courseAssessmentCount})',
+  },
+  achievementCountDetails: {
+    id: 'course.statistics.StatisticsIndex.course.StudentPerformanceTable.achievementCountDetails',
+    defaultMessage: 'No. of Achievements (Total: {courseAchievementCount})',
+  },
+  correctness: {
+    id: 'course.statistics.StatisticsIndex.course.StudentPerformanceTable.correctness',
+    defaultMessage: 'Correctness',
+  },
+  videoSubmissionCountHeader: {
+    id: 'course.statistics.StatisticsIndex.course.StudentPerformanceTable.videoSubmissionCountHeader',
+    defaultMessage: 'Videos Watched (Total: {courseVideoCount})',
+  },
+  videoPercentWatchedHeader: {
+    id: 'course.statistics.StatisticsIndex.course.StudentPerformanceTable.videoPercentWatchedHeader',
+    defaultMessage: 'Average Video % Watched',
+  },
+  noData: {
+    id: 'course.statistics.StatisticsIndex.course.StudentPerformanceTable.noData',
+    defaultMessage: 'No Data',
+  },
+  csvFileTitle: {
+    id: 'course.statistics.StatisticsIndex.course.csvFileTitle',
+    defaultMessage: 'Student Performance Statistics',
+  },
+  searchBar: {
+    id: 'course.statistics.StatisticsIndex.course.searchBar',
+    defaultMessage: 'Search by Student Name',
+  },
+});
+
+interface Props {
+  courseAchievementCount: number;
+  courseAssessmentCount: number;
+  courseVideoCount: number;
+  hasGroupManagers: boolean;
+  hasPersonalizedTimeline: boolean;
+  isCourseGamified: boolean;
+  maxLevel: number;
+  showVideo: boolean;
+  students: CourseStudent[];
+}
+
+const StudentPerformanceTable: FC<Props> = (props) => {
+  const {
+    students,
+    hasPersonalizedTimeline,
+    isCourseGamified,
+    showVideo,
+    courseVideoCount,
+    courseAchievementCount,
+    courseAssessmentCount,
+    maxLevel,
+    hasGroupManagers,
+  } = props;
+
+  const { t } = useTranslation();
+  const [highlightPercentage, setHighlightPercentage] = useState(5);
+
+  const numHighlightedStudents = Math.floor(
+    (students.length * highlightPercentage) / 100,
+  );
+
+  if (isCourseGamified) {
+    students.sort(
+      (s1, s2) => (s2.experiencePoints ?? 0) - (s1.experiencePoints ?? 0),
+    );
+  } else {
+    students.sort((s1, s2) => (s2.correctness ?? 0) - (s1.correctness ?? 0));
+  }
+
+  const studentsWithHighlight = students.map((student, index) => ({
+    ...student,
+    isTopStudent: index <= numHighlightedStudents,
+    isBottomStudent: index >= students.length - numHighlightedStudents,
+  }));
+
+  const getStudentHighlightColor = (student: CourseStudent): string => {
```

Comment:
Move this out from the component so it doesn't get re-declared everytime the component is re-rendered.

Reply:
done